app/src/main/java/org/mozilla/fenix/components/metrics/UsageRecorder.kt

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.metrics
+
+import android.app.Activity
+import android.content.Context
+import mozilla.components.support.utils.ext.getPackageInfoCompat
+import org.mozilla.fenix.android.DefaultActivityLifecycleCallbacks
+import org.mozilla.fenix.utils.Settings
+
+/**
+ * This will store app usage time to disk, based on Resume and Pause lifecycle events. Currently,
+ * there is only interest in usage during the first day after install.
+ */
+class UsageRecorder(private val settings: Settings) : DefaultActivityLifecycleCallbacks {
```

Comment:
We should add some basic tests to ensure we are saving the time as expected. Maybe something like this? Just an idea. 🙂 

```kotlin
@Test
fun theBestTest() {
  val startTime = System.currentTimeMillis()

  UsageRecorder.onActivityResumed()
  UsageRecorder.onActivityPaused()

  val stopTime = System.currentTimeMillis()
  
  val usageTimeGrowthData = getUsageTimeGrowthData()
  assertTrue(usageTimeGrowthData < stopTime - startTime)
}

```

Reply:
Good call, I had skipped it to avoid spinning up an actual activity and hadn't thought that step could be skipped.